### PR TITLE
Refactor main tachometer runner code into a class

### PR DESCRIPTION
**Pull request by aomarks**
The main tachometer runner logic had gotten pretty hard to read. It still could be better, but this PR refactors it into a class as a start on some better organization. No behavioral changes here, just moving existing code around into methods.

### CODE REVIEW

File: src/automatic.ts [hunk 1]
```diff
@@ -28,60 +28,158 @@ import {Server} from './server';
...
 }
 
-export async function automaticMode(
-    config: Config, servers: Map<BenchmarkSpec, Server>):
-    Promise<Array<ResultStatsWithDifferences>|undefined> {
-  let completeGithubCheck;
-  if (config.githubCheck !== undefined) {
-    completeGithubCheck = await github.createCheck(config.githubCheck);
+export class AutomaticMode {
+  private config: Config;
```
Comment (rictic) [inline src/automatic.ts:38]: Many of these fields can be made readonly as well

File: src/automatic.ts [hunk 2]
```diff
@@ -28,60 +28,158 @@ import {Server} from './server';
...
+                Math.max(0, Math.round((timeoutMs - elapsed) / 1000));
+            const mins = Math.floor(remainingSecs / 60);
+            const secs = remainingSecs % 60;
+            process.stdout.write(
+                `\r${spinner[run % spinner.length]} Auto-sample ${sample} ` +
+                `(timeout in ${mins}m${secs}s)` + ansi.erase.inLine(0));
+            specResults.get(spec)!.push(await this.takeSample(spec));
+          }
+        }
+      }
```
Comment (rictic) [inline src/automatic.ts:176]: Consider flipping this conditional to leave the main path through the code unindented

```suggestion
  private async takeAdditionalSamples() {
    const {config, specs, specResults} = this;
    if (config.timeout <= 0) {
      return;
    }
    console.log();
    const timeoutMs = config.timeout * 60 * 1000;  // minutes -> millis
    const startMs = Date.now();
    let run = 0;
    let sample = 0;
    let elapsed = 0;
    while (true) {
      if (horizonsResolved(this.makeResults(), config.horizons)) {
        console.log();
        break;
      }
      if (elapsed >= timeoutMs) {
        this.hitTimeout = true;
        break;
      }
      // Run batches of 10 additional samples at a time for more presentable
      // sample sizes, and to nudge sample sizes up a little.
      for (let i = 0; i < 10; i++) {
        sample++;
        for (const spec of specs) {
          run++;
          elapsed = Date.now() - startMs;
          const remainingSecs =
            Math.max(0, Math.round((timeoutMs - elapsed) / 1000));
          const mins = Math.floor(remainingSecs / 60);
          const secs = remainingSecs % 60;
          process.stdout.write(
            `\r${spinner[run % spinner.length]} Auto-sample ${sample} ` +
            `(timeout in ${mins}m${secs}s)` + ansi.erase.inLine(0));
          specResults.get(spec)!.push(await this.takeSample(spec));
        }
      }
    }
  }
```

File: src/automatic.ts [hunk 3]
```diff
@@ -28,60 +28,158 @@ import {Server} from './server';
...
   initialTabHandle: string;
 }
 
-export async function automaticMode(
-    config: Config, servers: Map<BenchmarkSpec, Server>):
-    Promise<Array<ResultStatsWithDifferences>|undefined> {
-  let completeGithubCheck;
-  if (config.githubCheck !== undefined) {
-    completeGithubCheck = await github.createCheck(config.githubCheck);
+export class AutomaticMode {
```
Comment (rictic) [inline src/automatic.ts:37]: AutomaticMode -> AutomaticModeRunner?

Comment (aomarks) [reply src/automatic.ts]: Done (just Runner since there's no other kind of runner, and renamed the file too).

File: src/automatic.ts (hunk 1 above)
Comment (aomarks) [reply src/automatic.ts]: Done.

File: src/automatic.ts (hunk 2 above)
Comment (aomarks) [reply src/automatic.ts]: Done.

**Pull request merged**